### PR TITLE
Describe socket recv buffer bug

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: src/bugs/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# The Great Chapter of Bugs
+
+During the development of Jellyfish, we encountered a wide variety of bugs which solving sometimes took us
```
Comment (daniel-jodlos) [inline src/bugs/README.md]:
```suggestion
During the development of Jellyfish, we encountered a wide variety of bugs that took weeks or months to solve.
```

File: src/bugs/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+days or weeks.
+This chapter collects the most interesting ones, explaining where was the problem and how we solved it.
+We hope that you will find here things that we couldn't find anywhere.
```
Comment (daniel-jodlos) [inline src/bugs/README.md]:
```suggestion
This chapter collects the most interesting ones, explaining where the problem was and how we solved it.
We hope that you will find here things that we couldn't find anywhere.

**WARNING!!!**: This chapter contains extremely disturbing content. If you have weak nerves, don't read.
```

File: src/bugs/socket_buffer.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+At some point in implementing simulcast, we started noticing very strange statistics in the webrtc-internals.
+On some of our machines, web browsers connected via our WebRTC SFU were dropping a lot of RTP packets and video frames.
+The QoE was terrible.
+The client was displaying something around 3 FPS.
+It didn't matter whether we were testing in a local or remote network.
+At the beginning, we thought that this is some kind of packet loss but we couldn't get how it might be possible
+in the local network.
+We entirely turned off simulcast, decreased resolution, changed codecs and downgraded web browsers.
+Nothing helped.
+The only thing we noticed was that the problem was not occurring on pretty strong machines like those with the 12th gen i9 processors or when we disabled TWCC.
```
Comment (daniel-jodlos) [inline src/bugs/socket_buffer.md]: Actually, I think the issue here was mostly related to webcam quality :stuck_out_tongue:

File: src/bugs/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# The Great Chapter of Bugs
```
Comment (daniel-jodlos) [inline src/bugs/README.md:1]: How about incorporating a :hankey: emoji somewhere around here?

**Review by daniel-jodlos — CHANGES_REQUESTED** [review]
Rage intensifies. I think I would add some screenshots from webrtc internals to really drive the point home

File: src/bugs/README.md (hunk 4 above)
Comment (mickel8) [reply src/bugs/README.md:1]: This might be too much :smiley:

Comment (mickel8) [reply src/bugs/README.md:1]: I mean, the transition from `days or weeks` to  `weeks or months` should be enough

File: src/bugs/socket_buffer.md (hunk 3 above)
Comment (mickel8) [reply src/bugs/socket_buffer.md]: Actually we can check that

**Comment by mickel8** [comment]
>  I think I would add some screenshots from webrtc internals to really drive the point home

I am not sure whether those screenshots are redable enough :thinking:

File: src/bugs/socket_buffer/socket_buffer.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+Below, there are some plots from webrtc-internals dump that show the issue:
+
+![Frames Received](./frames_received.png)
+![Freezes Count](./freezes_count.png)
+![Packets Lost](./packets_lost.png)
+
+## The explanation
+
+After implementing TWCC, a client can finally estimate its real bandwidth which in turn allows it to send video in
```
Comment (daniel-jodlos) [inline src/bugs/socket_buffer/socket_buffer.md]: Shouldn't we also mention 300 kbps assumed by Chrome in absence of TWCC?

**Review by daniel-jodlos — APPROVED** [review]

**Comment by daniel-jodlos** [comment]
> I am not sure whether those screenshots are redable enough thinking

For anyone who finds this chapter as a solution to their problem, I would assume they are already quite fluent with WebRTC debugging, so they should be fine.

**Pull request merged**